packages/konsole-app/src/panels/RecordPanel.tsx

```diff
@@ -54,6 +54,7 @@ export const RecordPanel = () => {
   const [records, setRecords] = useState<IRecord[]>([]);
   const [search, setSearch] = useState('');
   const [delayedSearch, setDelayedSearch] = useState(search);
+  const filteredRecords = records.filter(record => recordType === record.type || !recordType);
```

Comment:
this shouldn't go here, but in the RegistryClient's `queryRecords`

Reply:
But RegistryClient will make an unnecessary network call. I think it should only be responsible for low level querying

Reply:
nope;
1) when you have thousands or millions (whatever) of records do you still want to fetch full size of dataset
2) in each client you'd need to replicate the logic of filtering, ordering, paging....

Reply:
OK, I see that if paging is supported on the server side, we want to do this differently.